pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveORCBatchResolver.java

```diff
@@ -0,0 +1,257 @@
+import static org.apache.hawq.pxf.api.io.DataType.SMALLINT;
+import static org.apache.hawq.pxf.api.io.DataType.TEXT;
+import static org.apache.hawq.pxf.api.io.DataType.TIMESTAMP;
+import static org.apache.hawq.pxf.api.io.DataType.VARCHAR;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.sql.Timestamp;
+import java.sql.Date;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.common.type.HiveDecimal;
+import org.apache.hadoop.hive.ql.exec.vector.VectorizedRowBatch;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.DoubleWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hawq.pxf.api.OneField;
+import org.apache.hawq.pxf.api.OneRow;
+import org.apache.hawq.pxf.api.ReadVectorizedResolver;
+import org.apache.hawq.pxf.api.UnsupportedTypeException;
+import org.apache.hawq.pxf.api.io.DataType;
+import org.apache.hawq.pxf.api.utilities.ColumnDescriptor;
+import org.apache.hawq.pxf.api.utilities.InputData;
+import org.apache.hawq.pxf.api.utilities.Plugin;
+import org.apache.hawq.pxf.plugins.hive.utilities.HiveUtilities;
+import org.apache.hadoop.hive.serde2.*;
+import org.apache.hadoop.hive.serde2.objectinspector.*;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.*;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector.PrimitiveCategory;
+import org.apache.hadoop.hive.ql.exec.vector.*;
+
+@SuppressWarnings("deprecation")
+public class HiveORCBatchResolver extends Plugin implements ReadVectorizedResolver {
+
+    private static final Log LOG = LogFactory.getLog(HiveORCBatchResolver.class);
+
+    private List<List<OneField>> resolvedBatch;
+    private StructObjectInspector soi;
+
+    public HiveORCBatchResolver(InputData input) throws Exception {
+        super(input);
+        try {
+            soi = (StructObjectInspector) HiveUtilities.getOrcReader(input).getObjectInspector();
+        } catch (Exception e) {
+            LOG.error("Unable to create an object inspector.");
+            throw e;
+        }
+    }
+
+    @Override
+    public List<List<OneField>> getFieldsForBatch(OneRow batch) {
+
+        Writable writableObject = null;
+        Object fieldValue = null;
+        VectorizedRowBatch vectorizedBatch = (VectorizedRowBatch) batch.getData();
+
+        // Allocate empty result set
+        resolvedBatch = new ArrayList<List<OneField>>(vectorizedBatch.size);
+        for (int i = 0; i < vectorizedBatch.size; i++) {
+            ArrayList<OneField> row = new ArrayList<OneField>(inputData.getColumns());
+            resolvedBatch.add(row);
+            for (int j = 0; j < inputData.getColumns(); j++) {
+                row.add(null);
```

Comment:
probably not very efficient, it'd be better to have a template empty row that can be clone() using underlying Arrays.copy() or similar methods.